src/create-client.test.ts

```diff
@@ -0,0 +1,349 @@
+/**
+ * @jest-environment-options {"url": "https://example.com/callback?code=code_123"}
+ */
+
+import { createClient } from "./create-client";
+import { LoginRequiredError, RefreshError } from "./errors";
+import { mockLocation, restoreLocation } from "./testing/mock-location";
+import { storageKeys } from "./utils/storage-keys";
+import nock from "nock";
+
+describe("create-client", () => {
+  let client: Awaited<ReturnType<typeof createClient>>;
+
+  beforeEach(() => {
+    sessionStorage.removeItem(storageKeys.codeVerifier);
+    localStorage.removeItem(storageKeys.refreshToken);
+  });
+
+  afterEach(() => {
+    client?.dispose();
+
+    // Restore the original URL. JSDOM/Jest apparently doesn't do this between tests.
+    history.replaceState(
+      null,
+      "",
+      "https://example.com/callback?code=code_123",
+    );
+  });
+
+  const mockAccessToken = () =>
+    "." +
+    btoa(
+      JSON.stringify({
+        sid: "session_123abc",
+        iat: Date.now() / 1000,
+        exp: Date.now() / 1000 + 3600,
+      }),
+    );
+
+  describe("createClient", () => {
+    describe("with no `clientId` provided", () => {
+      it("throws an error", async () => {
+        await expect(createClient("")).rejects.toThrow(/Missing Client ID/);
+      });
+    });
+
+    describe("when the current route matches the redirect URI", () => {
```

Comment:
i think maybe it's more accurate to say 'when a code param is present` or something like that?

Reply:
Very true. Will update those test names to be more specific.